Add PR checkbox for PR title requirements (#3345) (#3346)
DataBiosphere/azul: .github/pull_request_template.md

Author

- [ ] PR title references issue

> - [ ] PR title matches issue title (preceded by `Fix: ` for bugs)   <sub>or there is a good reason why the PR title deviates from the issue title</sub>

Reviewer: ```suggestion
- [ ] PR title matches issue title (preceded by `Fix: ` for bugs)   <sub>or there is a good reason why they're different</sub>
```
